source/modulo_core/modulo_core/joint_positions_broadcaster.py

```diff
@@ -0,0 +1,44 @@
+from typing import List, Optional, Union
+
+from modulo_interfaces.msg import JointPositions, JointPositionsCollection
+from rclpy.node import Node
+from rclpy.qos import DurabilityPolicy, HistoryPolicy, QoSProfile
+
+
+class JointPositionsBroadcaster:
+    """
+    The JointPositionsBroadcaster is a TF2 style class that publishes a collection of JointPositions messages to 
+    the fixed /joint_positions topic.
+    """
+
+    def __init__(self, node: Node, qos: Optional[Union[QoSProfile, int]] = None):
+        """
+        Constructor of the JointPositionsBroadcaster.
+
+        :param node: The ROS2 node
+        :param qos: A QoSProfile or a history depth to apply to the subscriber
+        """
+        if qos is None:
+            qos = QoSProfile(depth=1, durability=DurabilityPolicy.TRANSIENT_LOCAL, history=HistoryPolicy.KEEP_LAST)
+        self._publisher = node.create_publisher(JointPositionsCollection, '/joint_positions', qos)
+        self._net_message = JointPositionsCollection()
+        self._names = set()
+
+    def send(self, joint_positions: Union[JointPositions, List[JointPositions]]):
```

Comment:
Note that here and in the listener, I'm just working with `modulo_interfaces.msg.JointPositions` and not with the `state_representation.JointPositions` because it's not as easy without templates. I think the translation is so easy that it doesn't really matter.